docs/reference/technology/connectivity/test-lora.md

```diff
@@ -0,0 +1,289 @@
+# Mbed LoRaWAN Solution
+
+## What is LoRaWAN & Why ?
+The basic difference between "internet" and "internet of things (IoT)" is that for IoT, there is just 'less of everything' available for a given network device: less memory, less processing capacity, less leg-room in terms of radio spectrum availability, less storage capacity and ofcourse less available power. These restrictions appear either because most of the "things" which will constitute IoT are battery powered and maximizing battery life-time is a priority, or they are supposed to be deployed in massive amounts, we are talking billions and trillions of tiny devices here.
+
+This drive to “do more with less” leads to constraints that limit the applicability of traditional cellular networks, as well as of technologies, such as WiFi, due to energy and scalability requirements. A new class of wireless networking protocols have emerged to tackle the challanges posed by IoT which is generally known as Low Power Wide Area Networks (LPWANs). 
+
+LoRaWAN is one such LPWAN protocol and it offers: substantialy large raio coverage, low adaptive data rates, low link budgets, fairness and co-existence using aggressive duty cycling, less prone to interference modulation scheme etc. Essentially, LoRaWAN is designed for the battery-operated devices with very few resources in hand and which are supposed to sent very short packets (datagrams) intermittently. The long-range and low-power nature of LoRaWAN makes it an interesting candidate for smart sensing technology in civil infrastructures (such as health monitoring, smart metering, environment monitoring, etc.), as well as in industrial applications.
+
+LoRaWAN technology operates in unlicensed radio spectrum abiding by the regional radio frequency regulations and requirements (frequency band of operation depends upon the region or country). Although the radio technology in LoRaWAN networks, simply known as LoRa is proprietary (Semtech holds the patent), LoRaWAN itself is an open wireless networking standard. 
+
+As LoRaWAN devices operate in unlicensed spectrum, the costs of operation drop massively and it opens up a door for anyone to come up and deploy their own network either as a service provider or as private operator. Sky is the limit here.
+
+## LoRaWAN Network Architecture
+ 
+ A LoRaWAN network consists of three basic network elements:
+ 
+ 1. Device
+ 2. Base station 
+ 3. Network Server
+ 
+ Base station is essentially transparent, its job is simply to speak LoRa with the devices in its coverage area. The real network control lies in the cloud, i.e., the Network Server. However, there are many solutions available where Network Server also resides on the base station, making a non-scalable private LoRaWAN network. 
```

Comment:
"there are many solutions available where Network Server also resides on the base station" -> put a big fat warning there that this is not LoRaWAN compliant.

Reply:
@janjongboom I figured what you are saying would eventually be specified. However I've never seen any specifications that define particulars about the network server or application server other then what is said in the LoRaWAN spec about encrypting mac and application data and providing app ID.

My last spec read through was  sometime in 2017 though. Could you point me to where this is specified?

Reply:
@janjongboom My thinking behind this is that in order to do edgework with LoRa you'll need a network server on the local network. In this case I'd think gateways could forward traffic to both the local network server and a cloud network server for 'telemetry' purposes.

Base on the above I think there are some scenario's where having a LoRaWAN gateway and a LoRaWAN network server residing on the same hardware might be convenient.
The local network server would then forward application traffic to an application server elsewhere in the network.

Feel free to continue this conversation elsewhere if you'd prefer. My email should be in my github profile.

Reply:
There is no concept of slave and master network servers in the spec. So if you have a network server residing on Base Station, that's the end of your network. It's a single star network. You can't interconnect various stars to make a big network. That's what I meant by being non-scalable.